src/aiko_services/main/scheme.py

```diff
@@ -48,12 +48,16 @@ def destroy_targets(self, stream):
 
     @classmethod
     def parse_data_url_path(cls, data_url):  # data_source or data_target
+        if data_url.startswith("data:"):
+            raise NotImplementedError("'data:' URLs do not have a path component")
         tokens = data_url.split("://")  # URL "scheme://path" or "path"
         path = tokens[0] if len(tokens) == 1 else tokens[1]
         return path
 
     @classmethod
     def parse_data_url_scheme(cls, data_url):  # data_source or data_target
+        if data_url.startswith("data:"):
+            return "data"
```

Comment:
The URL specification is "scheme:[//authority]path" and web browsers typically accept just "path" to mean "file:path" (with an empty authority).  Need to properly handle various combinations of URLs with "scheme:path", "scheme://path", "scheme://authority/path" or just "path".

Time for me to incorporate the standard Python URL parsing library function call.